lesson03/assignment/index.js

```diff
@@ -0,0 +1,144 @@
+//Get data.json into index.html
+
+//Assign info from data.json a variable, userProfiles. JSON info differs from JS arrays in syntax by wrapping entire array in single quotes and wrapping key/values in double quotes. JS seems to be able to read the JSON with the double quotes.
+const userProfiles = [{
+    "name": "Paolo Maldini",
+    "jobTitle": "Front End Developer",
+    "company": "Web Weavers",
+    "experience": "3 years",
+    "school": "University of Washington",
+    "major": "Marketing",
+    "email": "paolo@example.com",
+    "linkedInUrl": "paolo.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS", "JavaScript", "Node", "Express"
+    ]
+  },
+  {
+    "name": "Barbara Bonansea",
+    "jobTitle": "Software Engineer",
+    "company": "Excellence in the Cloud",
+    "experience": "12 years",
+    "school": "University of Southern California",
+    "major": "Computer Science",
+    "email": "barbara@example.com",
+    "linkedInUrl": "barbara.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "JavaScript", "C", "Go"
+    ]
+  },
+  {
+    "name": "Javier Hernandez",
+    "jobTitle": "User Experience Engineer",
+    "company": "Web Sites and More",
+    "experience": "5 years",
+    "school": "Seattle University",
+    "major": "Performing Arts",
+    "email": "javier@example.com",
+    "linkedInUrl": "javier.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS"
+    ]
+  },
+  {
+    "name": "Maribel Dominguez",
+    "jobTitle": "Front End Engineer",
+    "company": "Bits and Bytes",
+    "experience": "6 years",
+    "school": "University of Washington",
+    "major": "Mechanical Engineering",
+    "email": "maribel@example.com",
+    "linkedInUrl": "maribel.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS", "JavaScript", "React", "Vue", "Redux"
+    ]
+  }
+]
+
+// select DOM node
+const userProfileContainer = document.querySelector(".template-hook");
+
+// Update userProfiles entry counter
+let i =+ 0;
+
+//============forEach DOM NODE FUNCTION============
+//run a forEach on userProfiles array
+userProfiles.forEach(function(el) { //passes each property name/value pair from userProfiles' array of objects to the el paramenter as arguement to function below.
+  //Create new HTML section
+  let nametag = document.createElement('section')
+  nametag.innerHTML = `
+  <h1>${el.name}</h1>
+  `
+
+  //==========Populate nametag section with object properties values from userProfiles.=========
+  //For below, can you automatically ouput property names too?
```

Comment:
There's a special kind of loop syntax for objects, so once you were inside the forEach loop, where `el` represents an object, you could then loop through `el` using for...in (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in). That would allow you to have access to each property of the object without needing to know in advance the name of each property. I think that's what you're asking, but if I'm misunderstanding, please clarify with a follow up email :)